pkg/cmd/server/api/types.go

```diff
@@ -83,6 +95,8 @@ const (
 	ControllersAll = "*"
 )
 
+type FeatureSet map[OpenShiftFeature]util.Empty
```

Comment:
I think I'd make `FeatureSet` a `[]string` that contains a `Has` method.  That way json serialization is still pretty and you don't lose ordering between internal and external formats.  Otherwise, people will see the order of their options change during roundtripping.